sdk/storage/azure-storage-blob/azure/storage/blob/aio/download_async.py

```diff
@@ -170,56 +109,18 @@ def _do_optimize(self, given_range_start, given_range_end):
         return chunk_data
 
 
-class StorageStreamDownloader(object):  # pylint: disable=too-many-instance-attributes
-    """A streaming object to download from Azure Storage.
-
-    The stream downloader can iterated, or download to open file or stream
-    over multiple threads.
-    """
+class _AsyncChunkIterator(object):
+    """Async iterator for chunks in blob download stream."""
 
-    def __init__(
-            self,
-            clients=None,
-            config=None,
-            offset=None,
-            length=None,
-            validate_content=None,
-            encryption_options=None,
-            **kwargs):
-        self.clients = clients
-        self.config = config
-        self.offset = offset
-        self.length = length
-        self.validate_content = validate_content
-        self.encryption_options = encryption_options or {}
-        self.request_options = kwargs
-        self.location_mode = None
-        self._download_complete = False
-        self._current_content = None
-        self._iter_downloader = None
+    def __init__(self, size, content, downloader):
+        self.size = size
+        self._current_content = content
+        self._iter_downloader = downloader
         self._iter_chunks = None
-
-        # The service only provides transactional MD5s for chunks under 4MB.
-        # If validate_content is on, get only self.MAX_CHUNK_GET_SIZE for the first
-        # chunk so a transactional MD5 can be retrieved.
-        self.first_get_size = self.config.max_single_get_size if not self.validate_content \
-            else self.config.max_chunk_get_size
-        initial_request_start = self.offset if self.offset is not None else 0
-        if self.length is not None and self.length - self.offset < self.first_get_size:
-            initial_request_end = self.length
-        else:
-            initial_request_end = initial_request_start + self.first_get_size - 1
-
-        self.initial_range, self.initial_offset = process_range_and_offset(
-            initial_request_start, initial_request_end, self.length, self.encryption_options)
-        self.download_size = None
-        self.file_size = None
-        self.non_empty_ranges = None
-        self.response = None
-        self.properties = None
+        self._complete = (size == 0)
 
     def __len__(self):
-        return self.download_size
+        return self.size
 
     def __iter__(self):
```

Comment:
>): [](start = 21, length = 2)

then do we also need __next__ to remind the user using anext() instead?

Reply:
There is no `anext` builtin - so I don't think this will be a problem.